### PR TITLE
[#5217] docs(metalake-catalog): refine docs for deleting metalake and catalog

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

refine docs for deleting metalake and catalog

### Why are the changes needed?

Fix: #5217 

### Does this PR introduce _any_ user-facing change?

yes, user docs changed

### How was this patch tested?

setup the gravitino-site locally

### CODE REVIEW

File: docs/manage-messaging-metadata-using-gravitino.md [hunk 1]
```diff
@@ -138,7 +138,7 @@ in relational catalog for more details. For a messaging catalog, the list operat
 ## Topic operations
 
 :::tip
-Users should create a metalake, a catalog and a schema before creating a table.
+Users should create a metalake, a catalog and a schema, then ensure that the metalake and catalog are enabled before performing topic operations.
```
Comment (jerryshao) [inline docs/manage-messaging-metadata-using-gravitino.md:141]: "...before operating topics."

File: docs/manage-metalake-using-gravitino.md [hunk 2]
```diff
@@ -151,9 +151,108 @@ The following table outlines the supported modifications that you can make to a
...
+```python
+gravitino_admin_client: GravitinoAdminClient = GravitinoAdminClient(uri="http://localhost:8090")
+gravitino_admin_client.enable_metalake("metalake")
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+If the metalake is already enabled, the operation will do nothing.
```
Comment (jerryshao) [inline docs/manage-metalake-using-gravitino.md:194]: "This operation does nothing if the metalake is already enabled."

File: docs/manage-metalake-using-gravitino.md [hunk 3]
```diff
@@ -151,9 +151,108 @@ The following table outlines the supported modifications that you can make to a
...
+:::
+
+## Disable a metalake
+
+Once a metalake is disabled:
+ - It can only be [listed](#list-all-metalakes), [loaded](#load-a-metalake), [dropped](#drop-a-metalake), or [enabled](#enable-a-metalake).
```
Comment (jerryshao) [inline docs/manage-metalake-using-gravitino.md:200]: "Users can only xxx, xx, xxx".

File: docs/manage-relational-metadata-using-gravitino.md [hunk 4]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+
+The following is an example of enabling a catalog:
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PATCH -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{"in-use": true}' \
+http://localhost:8090/api/metalakes/metalake/catalogs\catalog
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:196]: `catalogs\catalog`, what is the meaning here?

File: docs/manage-relational-metadata-using-gravitino.md [hunk 5]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+// Assuming you have created a metalake named `metalake` and a catalog named `catalog`
+gravitinoClient.enableCatalog("catalog");
+// ...
+```
+
+</TabItem>
+</Tabs>
+
+:::info
+If the catalog is already enabled, the operation will do nothing.
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:213]: Ditto.

File: docs/manage-relational-metadata-using-gravitino.md [hunk 6]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+:::
+
+### Disable a catalog
+
+Once a catalog is disabled:
+- It can only be [listed](#list-all-catalogs-in-a-metalake), [loaded](#load-a-catalog), [dropped](#drop-a-catalog), or [enabled](#enable-a-catalog).
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:219]: Ditto.

File: docs/manage-relational-metadata-using-gravitino.md [hunk 7]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+
+The following is an example of disabling a catalog:
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PATCH -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{"in-use": false}' \
+http://localhost:8090/api/metalakes/metalake/catalogs\catalog
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:232]: Also here.

File: docs/manage-relational-metadata-using-gravitino.md [hunk 8]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+
+```java
+// ...
+// Assuming you have created a metalake named `metalake` and a catalog named `catalog`
+gravitinoClient.disableCatalog("catalog");
+// ...
+```
+
+</TabItem>
+</Tabs>
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:246]: Why there's no Python API here?

File: docs/manage-relational-metadata-using-gravitino.md [hunk 9]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+
+:::info
+If the catalog is already disabled, the operation will do nothing.
+:::
+
 ### Drop a catalog
 
+Deleting catalog doesn't use forced deletion by default, so please make sure:
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:254]: "Deleting a catalog by "force" is not a default behavior, so please..."

File: docs/manage-relational-metadata-using-gravitino.md [hunk 10]
```diff
@@ -180,8 +180,88 @@ Therefore, do not change the catalog's URI unless you fully understand the conse
...
+
+- There are no schemas under the catalog. Otherwise, you will get an error.
+- The catalog is [disabled](#disable-a-catalog). Otherwise, you will get an error.
+
+If you are using forced deletion, it will:
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:259]: "Deleting a catalog by "force" will:"

Comment (jerryshao) [reply docs/manage-relational-metadata-using-gravitino.md:259]: Can you please also reword the related part in metalake.

File: docs/open-api/catalogs.yaml [hunk 11]
```diff
@@ -201,11 +201,47 @@ paths:
         "5xx":
           $ref: "./openapi.yaml#/components/responses/ServerErrorResponse"
 
+    patch:
+      tags:
+        - catalog
+      summary: set caatalog in-use
```
Comment (jerryshao) [inline docs/open-api/catalogs.yaml:207]: "catalog"

File: docs/manage-relational-metadata-using-gravitino.md (hunk 8 above)
Comment (jerryshao) [reply docs/manage-relational-metadata-using-gravitino.md:246]: This issue is not addressed.

File: docs/manage-relational-metadata-using-gravitino.md [hunk 12]
```diff
@@ -268,7 +345,7 @@ Catalog[] catalogsInfos = gravitinoMetaLake.listCatalogsInfo();
 ## Schema operations
 
 :::tip
-Users should create a metalake and a catalog before creating a schema.
+Users should create a metalake and a catalog, then ensure that the metalake and catalog are enabled before performing schema operations.
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:348]: "before operating schemas."

File: docs/manage-relational-metadata-using-gravitino.md [hunk 13]
```diff
@@ -518,7 +595,7 @@ schema_list: List[NameIdentifier] = catalog.as_schemas().list_schemas()
 ## Table operations
 
 :::tip
-Users should create a metalake, a catalog and a schema before creating a table.
+Users should create a metalake, a catalog and a schema, then ensure that the metalake and catalog are enabled before performing table operations.
```
Comment (jerryshao) [inline docs/manage-relational-metadata-using-gravitino.md:598]: "before operating tables."

File: docs/manage-relational-metadata-using-gravitino.md (hunk 8 above)
Comment (mchades) [reply docs/manage-relational-metadata-using-gravitino.md:299]: added

**Review by jerryshao — APPROVED** [review]

**Pull request merged**